app/services/container_dashboard_service.rb

```diff
@@ -95,6 +96,30 @@ def providers
     result.values
   end
 
+  def alerts
+    if @ems.present?
+      warnings = @ems.miq_alert_statuses.where(:severity => "warning").count
+      errors = @ems.miq_alert_statuses.where(:severity => "error").count
+    else
+      warnings = MiqAlertStatus.where.not(:ems_id => nil).where(:severity => "warning").count
+      errors = MiqAlertStatus.where.not(:ems_id => nil).where(:severity => "error").count
+    end
```

Comment:
```
relation = @ems.present? ? @ems : MiqAlertStatus.where.not(:ems_id => nil)
warnings, errors = relation.group(:severity).values_at(*%w(warning error))
```
something like this to cut the number of SQL queries (and also source lines) by 2?